[DOCS] DOC-349 re-write and partition interactive mode expectations guide

Rachel-Reverie opened the pull request:
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- Significantly re-writes and re-organizes the "Create and edit Expectations in Interactive Mode" guide
- Partitions each section of the "Create and edit Expectations in Interactive Mode" guide


After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-


### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation



Thank you for submitting!

Rachel-Reverie committed:
DOC-349
- Significantly re-writes and re-organizes the "Create and edit Expectations in Interactive Mode" guide
- Partitions each section of the "Create and edit Expectations in Interactive Mode" guide

Rachel-Reverie committed:
DOC-349
- Significantly re-writes and re-organizes the "Create and edit Expectations in Interactive Mode" guide
- Partitions each section of the "Create and edit Expectations in Interactive Mode" guide

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=5448&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=5448&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/5448/be803b21/e3b1194995745c4dfebf0c30c1a448ead47f89f9.svg)](https://app.codesee.io/r/reviews?pr=5448&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

rdodev commented on docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_optional_edit_an_existing_expectation_suite_in_interactive_mode.mdx:
```diff
@@ -0,0 +1,20 @@
+
+This will open a Jupyter Notebook that will show the Expectations currently configured for your Expectation Suite.  You can then add, delete, or otherwise edit these Expectations.  When you are done, you will simply save the Expectation Suite and overwrite the old Expectations with the new ones you have executed.
+
+:::tip
```
I'm a bit confused here. Above we're instructing the user how to edit an existing expectation suite, and in the note below we tell them how to create a new one? It seems to me like two different tasks/workflows that should be in their own section/doc instead of an admonition.

Rachel-Reverie replied:
I thought it was a kind of odd tip to include in the original doc, myself, but didn't want to cut it out in case it was part of a use case I wasn't familiar with.  It seemed like the best spot for it at the time was here, as a "if you don't have an existing Expectation, you can make one like this" sort of insert, but I agree that the `--no-jupyter` flag is an odd inclusion.  Since using `--no-jupyter` actively prevents you from entering the Interactive workflow unless you then edit newly created expectation with the command above, this seemed like the best place to keep it.

Rachel-Reverie replied:
I've removed this particular tip (and temporarily copied it into a file in our docs_pending folder, so it can be included in another doc when a more appropriate one is identified).

rdodev commented on docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_specify_a_datasource_if_multiple_are_available.mdx:
```diff
@@ -0,0 +1,3 @@
+
+Next, the CLI will determine which Data Asset you intend to run your new Expectations against.  If you only have one configured Datas Asset, it will do so automatically.  If there are multiple options, you will be prompted to choose one.  If you do not have a Datasource configured, the CLI exit after alerting you to the issue.  You need to have a Datasource configured to continue the Interactive  ...
```
Minor typo? Data~s~ Asset

Rachel-Reverie replied:
Typo.  I've pushed a fix.

Rachel-Reverie committed:
DOC-349
- Typo correction

Rachel-Reverie committed:
DOC-349
- Typo correction

Rachel-Reverie committed:
DOC-349
- Removed a tip that seemed inappropriate to the workflow; put it temporarily in docs_pending/_tips so that it can be included elsewhere when a more appropriate doc is identified for it.

Rachel-Reverie committed:
DOC-349
- Removed a tip that seemed inappropriate to the workflow; put it temporarily in docs_pending/_tips so that it can be included elsewhere when a more appropriate doc is identified for it.

rdodev reviewed (approved):
LGTM!

Rachel-Reverie committed:
DOC-357
- Partitions each section of the "How to create a new Checkpoint" guide.
- Performs some updates to the content and structure of the "How to create a new Checkpoint" guide (mostly to the header names and organization).

Rachel-Reverie committed:
-correction to relative path for link

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/DOCS/DOC-280/DOC-349/partition_create_expectations_interactively_guide' into DOCS/DOC-280/DOC-349/partition_create_expectations_interactively_guide

# Conflicts:
#	docs/guides/expectations/components_how_to_create_and_edit_expectations_with_instant_feedback_from_a_sample_batch_of_data/_optional_profile_your_data_to_generate_expectations_then_edit_them_in_interactive_mode.mdx

Rachel-Reverie committed:
-correction to relative path for link

Rachel-Reverie committed:
Merge branch 'develop' into DOCS/DOC-280/DOC-349/partition_create_expectations_interactively_guide
